### PR TITLE
Screenshot: Add Coupon Card to the My Store section of Screenshot tool 

### PR DESCRIPTION
**Pull request by hafizrahman**
## Description

When generating screenshots on iPad device, previously My Store area might look a bit too empty because there's not enough cards on it.

This quick PR adds the Most Active Coupon card to the My Store are to fill in the space.

At the moment, the Coupon card is shown in empty state. I tried making it show mocked data, but it's non-trivial, because:

1. The card fetches data from storage based on siteID value,
2. The mocked Coupon can't have its siteID value set manually, it has to be set by its Mapper after retrieving network data:

https://github.com/woocommerce/woocommerce-ios/blob/89297f90de3c9248cbc403bedac8d599e4ce371b/Networking/Networking/Model/Coupon.swift#L11-L12

If you have suggestion on how to workaround that limitation, do let me know! At the moment I suppose it's OK because the Coupon card is not even shown fully. It's just there to help fill in the space.

## Steps to reproduce

1. Set target to `WooCommerceScreenshot` and pick a tablet device (I'm using iPad Pro 12.0 2nd gen with iOS 16)
2. Build the app,
3. Compare with screenshot below in "After"

To check the "Before" screenshot, repeat the steps above on the `trunk` branch.

## Testing information

I tested this in Simulator with iPad Pro 12.0 2nd gen with iOS 16

## Screenshots

| Before | After |
|-|-|
| Simulator Screenshot - ScreenShot iPad Pro 12 0 2nd Gen iOS 16 - 2024-11-21 at 12 24 19 | Simulator Screenshot - ScreenShot iPad Pro 12 0 2nd Gen iOS 16 - 2024-11-21 at 12 21 38 |

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/232">21.2</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Yosemite/Yosemite/Model/Mocks/ActionHandlers/MockCouponActionHandler.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import Storage
+import Networking
+
+struct MockCouponActionHandler: MockActionHandler {
+
+    typealias ActionType = CouponAction
+
+    let objectGraph: MockObjectGraph
+    let storageManager: StorageManagerType
+    let store: CouponStore
```
Comment (selanthiraiyan) [inline Yosemite/Yosemite/Model/Mocks/ActionHandlers/MockCouponActionHandler.swift]: nit: I think we can remove this `store` as we don't use it.

**Review by selanthiraiyan — APPROVED** [review]
Thanks for working on this! LGTM. :shipit:

File: Yosemite/Yosemite/Model/Mocks/ActionHandlers/MockCouponActionHandler.swift (hunk 1 above)
Comment (hafizrahman) [reply Yosemite/Yosemite/Model/Mocks/ActionHandlers/MockCouponActionHandler.swift]: Thanks! https://github.com/woocommerce/woocommerce-ios/pull/14477/commits/5cffd2c3f4260fbc8e1123204c77435a330e4756

**Comment by hafizrahman** [comment]
I added an extra commit to update the release note about fixing `WooCommerceScreenshot`, as the original PR didn't include it.

**Pull request merged**